org.eclipse.jdt.compiler.apt/src/org/eclipse/jdt/internal/compiler/apt/dispatch/AjBatchFilerImpl.java

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2014 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Sergey Stupin - contributions for adding annotation processing support to ajc
+ *******************************************************************************/
+package org.eclipse.jdt.internal.compiler.apt.dispatch;
+
+import org.eclipse.jdt.internal.compiler.batch.CompilationUnit;
+
+import javax.lang.model.element.Element;
+import javax.tools.FileObject;
+import javax.tools.JavaFileManager;
+import java.io.*;
+import java.net.URI;
+
+/**
+ * @author s.stupin
+ */
+public final class AjBatchFilerImpl extends BatchFilerImpl {
```

Comment:
Confirmed. I also wrote you some lengthy e-mail a few weeks ago, explaining that I did it and why: in order to enable easy up-stream merges, because in the old "shadows" repository the files from the original module were just copied into the other module. That does not merge well (as in: not at all) with Git. I also explained that I used a virtual folder in the updated [Eclipse project settings](https://github.com/kriegaex/aspectj.eclipse.jdt.core/blob/c04eff5c2571963c4cb6a71e1f1cbce3a8987aed/org.eclipse.jdt.core/.project#L18-L24) in order to emulate `apt` being a local folder inside the APT Core module we are mainly working in, so that should feel pretty much like before to Eclipse users (I work in IntelliJ IDEA for the most part, though).